src/app/frontend/common/autorefresh/poll_service.js

```diff
@@ -0,0 +1,208 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Service class for polling.
+ * @final
+ */
+export class PollService {
+  /**
+   * @param $interval {!angular.$interval}
+   * @param $q {!angular.$q}
+   * @ngInject
+   */
+  constructor($interval, $q) {
+    /**
+     * @type {!angular.$interval}
+     * @private
+     */
+    this.intervalService = $interval;
+
+    /**
+     * @type {!angular.$q}
+     * @private
+     */
+    this.q_ = $q;
+
+    /**
+     * @type {!Array<PollOptions>}
+     * @private
+     */
+    this.optionsList;
+
+    /**
+     * @type {!PollOptions}
+     */
+    this.currentOptions;
+  }
+
+  /**
+   * Starts polling and returns polling promise.
+   * @param resource {!angular.$resource}
+   * @param params {angular.Resource.ParamsOrCallback=}
+   * @param delay {number}
+   * @returns {!angular.$q.Promise}
+   */
+  poll(resource, params, delay) {
```

Comment:
Thank you for spotting the bug! It wasn't working as intended. I've created new method for pollService to remove all optionsList on state change. It was designed to keep multiple running polls on the same state. In this case it was holding the options from previous state which had to be removed from the list.